mirrord/layer/src/socket/hooks.rs

```diff
@@ -370,6 +371,128 @@ pub(super) unsafe extern "C" fn send_to_detour(
         })
     }
 }
+
+#[hook_guard_fn]
+pub(super) unsafe extern "C" fn recvmsg_detour(
+    sockfd: i32,
+    message_header: *mut libc::msghdr,
+    flags: c_int,
+) -> ssize_t {
+    // Returns `-1`, but calling the original function sets whatever `errno` for us.
```

Comment:
I think you can just call recvmsg on the socket, then just modify the msg headers.. no?